src/components/filterBox/FilterBoxActions.ts

```diff
@@ -6,18 +7,35 @@ export const FilterActions = {
     filterThrough: 'FILTER',
 };
 
+export const defaultMatchFilter = (filterValue: string, item: any, propertiesName: string[]) => {
+    if (filterValue === '') {
+        return true;
+    }
+
+    if (filterValue && item) {
+        const regex = new RegExp(filterValue, 'gi');
+        return _.some(propertiesName, (property: string) => item[property] ? regex.test(item[property]) : false);
+    }
+
+    return false;
+};
+
+export type MatchFilter = (filterValue: string, item: any, propertiesName: string[]) => boolean;
+
 export interface IFilterActionPayload {
     id: string;
+    matchFilter?: MatchFilter;
 }
 
 export interface IChangeFilterActionPayload extends IFilterActionPayload {
     filterText: string;
 }
 
-export const addFilter = (id: string): IReduxAction<IFilterActionPayload> => ({
+export const addFilter = (id: string, matchFilter: MatchFilter = defaultMatchFilter): IReduxAction<IFilterActionPayload> => ({

```

Comment:
If `defaultMatchFilter`'s type is defined properly, typescript can guess the type by the given default value, so no need to type directly `matchFilter`